src/module.c

```diff
@@ -3151,7 +3176,9 @@ int RM_StreamAdd(RedisModuleKey *key, int flags, RedisModuleStreamID *id, RedisM
         errno = EDOM;
         return REDISMODULE_ERR;
     }
-    signalKeyAsReady(key->db, key->key, OBJ_STREAM);
+    /* Postponed signalKeyAsReady(). Done implicitly by moduleCreateEmptyKey()
+     * so not needed if the stream has just been created, i.e. if length == 1 */
+    key->u.stream.signalready = (s->length > 1);
```

Comment:
don't you mean `|=`?

Reply:
Right... But not even that. The stream can also be empty and when adding one it will have length == 1. I'll have to rework this a little.

Reply:
Changed to `if (!created) key->u.stream.signalready = 1;`